### PR TITLE
Feature 4: Recipe Searcher

### PR DESCRIPTION
**Pull request by Ge0n0**
Overwrote equals() method in Ingredient, Tool, and Tag classes.
searchRecipe method in RecipeSearcher works only when one of either Ingredient, Tool, or Tag is passed to the method. added Tests for searchRecipe method which checks when one item is passed to the method.

### CODE REVIEW

File: src/main/java/entity/Tool.java [hunk 1]
```diff
@@ -34,4 +34,11 @@ public String getName() {
     public void setName(String name) {
         this.name = name;
     }
+
+    public boolean equals(Tool t){
```
Comment (lukekrikorian) [inline src/main/java/entity/Tool.java:38]: Should this be marked as `@Override`?

File: src/main/java/entity/Tag.java [hunk 2]
```diff
@@ -34,4 +34,11 @@ public String getName() {
     public void setName(String name) {
         this.name = name;
     }
+
+    public boolean equals(Tag t){
```
Comment (lukekrikorian) [inline src/main/java/entity/Tag.java:38]: Should this be marked as `@Override`?

File: src/main/java/entity/Ingredient.java [hunk 3]
```diff
@@ -32,4 +32,11 @@ public String getName() {
     public void setName(String name) {
         this.name = name;
     }
+
+    public boolean equals(Ingredient i){
```
Comment (lukekrikorian) [inline src/main/java/entity/Ingredient.java:36]: I think this should be marked `@Override` as well

**Review by lukekrikorian — CHANGES_REQUESTED** [review]
All the code looks good to me!

There are a few formatting issues in that could be resolved by running `Code > Reformat File...`  with all the options (optimize imports, code cleanup, rearrange code, do not keep line breaks) turned on. I'd recommend running this on the files changed by this commit and make another commit with the formatting fixes.

File: src/main/java/entity/Ingredient.java (hunk 3 above)
Comment (Selenarn) [reply src/main/java/entity/Ingredient.java:36]: I'm a little confused. In my opinion, they are not subclass and parent classes so they don't need to mark Override, is that correct? Though they have the same name, they are not replacing each other, right?

Comment (lukekrikorian) [reply src/main/java/entity/Ingredient.java:36]: They're overriding `Class`'s default `equals` method.

**Review by Cbwww666 — APPROVED** [review]

**Review by lukekrikorian — APPROVED** [review]
Looks good!

**Pull request merged**